build_docs.sh

```diff
@@ -16,7 +16,8 @@ cd ${WSDIR}
 mkdir src
 wget -O ros2.repos https://raw.githubusercontent.com/ros2/ros2/${RELEASE_NAME}-release/ros2.repos
 vcs import src < ros2.repos
-colcon build --packages-up-to class_loader geometry2 rclcpp_action rclcpp_components rclcpp_lifecycle rclpy ament_index_cpp ament_index_python libstatistics_collector
+git clone https://github.com/ros/console_bridge/ src/ros/console_bridge
```

Comment:
Until now, documentation generation has been based on a release repos file. Cloning a repo's latest changes means that it's possible we will generate incorrect documentation for a given release. E.g. if the version of `console_bridge` used in Dashing is different than Foxy, then invoking this script for Dashing will generate the wrong version of documentation. (I don't actually know if there are significant changes in `console_bridge` between Dashing and Foxy, but I think it's something we should be careful about.) I don't think we want to assume that this script is only ever run for the latest ROS 2 release, or the latest state of master.

If we really want to include `console_bridge` in the documentation generation process, then I suggest doing it in a way where we can clone the correctly tagged version for a given ROS 2 code name. I can think of a couple ways:

1. Maintain a separate set of release repos file in this repository for supplemental repositories. e.g. `dashing.repos`, `eloquent.repos`, etc, each containing the information for `console_bridge`.

2. Use the `rosdistro` Python API to query the correct version for a given ROS distribution.

Reply:
Thinking about it some more, `console_bridge` is treated as a dependency provided by the platform distribution. Considering how it is further removed from the ROS ecosystem, I'm not sure we should be responsible for generating and uploading it's documentation. For instance, it doesn't even show up on ROS Index ([search](https://index.ros.org/search/?term=console_bridge)).

cc/ @scpeters if you have any thoughts.

Reply:
to get console_bridge to quality level 1, we want it to have documentation somewhere, I guess the question is whether we host it with existing ROS infrastructure or do something else

Reply:
I'm hesitant about hosting with the ROS infrastructure, because it could easily disappear when we get around to adding infrastructure for proper doc jobs. This is because `console_bridge` does not have an entry in the [distribution.yaml](https://github.com/ros/rosdistro/blob/master/foxy/distribution.yaml) for any ROS distro since Bouncy. `console_bridge` is a rosdep; it seems odd to think that we would generate and host docs for rosdep entries. Maybe we can consider updating the [doc entry for `console_bridge_vendor`](https://github.com/ros/rosdistro/blob/master/foxy/distribution.yaml#L288-L291) to point to the `console_bridge` repository. Then once we start using the `doc` entries, the `console_bridge` API docs will be generated and hosted under the name `console_bridge_vendor`. If we do this, then I'm okay with adding `console_bridge` to this script with a way to align the version with the ROS distro.

Reply:
uhmm, `console_bridge_vendor` has a cmake option `FORCE_BUILD_VENDOR_PKG` then in the `build` directory the source code is downloaded.

what about building the doxygen documentation from this folder?

Reply:
ping @ahcorde

Reply:
@jacobperron I bumped the version of `console_bridge_vendor` to `1.0.1` on `master` but the foxy banch is targeting `console_bridge_vendor` 0.5.1, should I backport this change? [There are not ABI and/or API issues](https://github.com/ros/console_bridge/commits/master)

Reply:
If there's no significant code changes, feel free to open a backport PR if you want those changes in Foxy.

Reply:
@jacobperron CI is green!